src/ipips/ipip-0445.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "IPIP-0445: trustless gateway skip-leaves option"
+date: 2023-10-09
+ipip: open
+editors:
+  - name: Hugo VALTIER
+    github: Jorropo
+    url: https://jorropo.net/
+    affiliation:
+        name: Protocol Labs
+        url: https://protocol.ai/
+relatedIssues:
+  - https://github.com/ipfs/specs/issues/444
+order: 445
+tags: ['ipips']
+---
+
+## Summary
+
+Introduce `skip-leaves` flag for the :cite[trustless-gateway].
+
+## Motivation
+
+Allow clients to read a stream which only contains the unixfs proofs.
```

Comment:
I'd like to see some details in here that address the non-unixfs impact of this, or at least make it clear that this is compatible with non-unixfs graphs (or not?) since our current trustless design only special-cases unixfs, it doesn't assume it; this is written with what sounds like an assumption and could be adjusted.

Reply:
This is the `Motivation` field, I don't think it need to be an accurate depiction of all the possible usecases.
I don't have any non-unixfs motivations, could you give me an example of an other data-structure which can use raw cids for the bulk of their data in a way that would be interesting here ? (bottom heavy graphs, download candidates discovery for bitswap, ...)

Reply:
sorry, I just tagged here because it's the first mention of `unixfs`, I just think discussion would be good so we're clear that this is interoperable with other types of graphs.

two things come to mind:

1. you could just be clear that this would still work with non-unixfs, it's entirely dependent on codec in the CID
2. there are some novel cases where you could _use_ this to provide alternate addressing of existing data. Perhaps you have a standard unixfs graph but it's deep and complicated, so just bundle it up in some custom dag-cbor, linking to all the `raw` blocks, and you have a custom kind of proof pointing to the same file data; i.e. this could unlock custom proof and/or addressing mechanisms .. maybe (not claiming it will, or even that I think this is worthwhile, just talking off the top of my head).

Reply:
I see thx, I've updated the text, first sentence don't mention unixfs and I've moved it to the sentence where I link #444